notebook/services/contents/filemanager.py

```diff
@@ -183,6 +183,14 @@ def is_hidden(self, path):
         os_path = self._get_os_path(path=path)
         return is_hidden(os_path, self.root_dir)
 
+    def _get_file_size(self, path):
+        try:
+            size = os.path.getsize(path)
```

Comment:
This will do a new stat on the file, which is best avoided for performance reasons. We already get a stat object in `_base_model()` - perhaps we can use the size from that, and remove it again for directories. It's a bit of an ugly trick, but for a big directory, it could make a difference.

Reply:
@takluyver thank you -- I've updated so that it uses file size from `st_size` (retrieved from stat) in `_base_model()`, but let me know if this isn't what you meant